sdk/eventhub/Azure.Messaging.EventHubs/tests/Consumer/EventHubConsumerClientLiveTests.cs

```diff
@@ -2585,8 +2600,14 @@ Task<bool> readCallback(ReadState currentState)
         ///
         private class ReadState
         {
-            public readonly ConcurrentDictionary<string, PartitionEvent> Events = new ConcurrentDictionary<string, PartitionEvent>();
+            public readonly HashSet<string> RemainingEvents;
+            public readonly ConcurrentDictionary<string, PartitionEvent> Events = new();
             public long EmptyCount = 0;
+
+            public ReadState(IEnumerable<string> expectedEvents) =>
```

Comment:
nit: Unless your tests require that this collection be copied, you could check for `is HashSet<string> hash` and assign it directly to `RemainingEvents` (or just create an additional ctor overload).

Reply:
The copying is intentional in this case; I need an independent set for each insanitation.   Some of the tests have multiple readers working against the same set of events, where each is keeping track of which they haven't yet received.